tests/MongoDB.Driver.Tests/Specifications/crud/FindOneAndUpdateTest.cs

```diff
@@ -81,11 +88,31 @@ protected override BsonDocument ExecuteAndGetResult(IMongoCollection<BsonDocumen
         {
             if (async)
             {
-                return collection.FindOneAndUpdateAsync(_filter, _update, _options).GetAwaiter().GetResult();
+                if (_updatePipeline != null)
```

Comment:
If you add these lines at the beginning of the method the other lines don't need any changes other than replacing `_update` with `updateDefinition`:

    UpdateDefinition<BsonDocument> updateDefinition = null;
    if (_update is BsonDocument updateDocument)
    {
        updateDefinition = new BsonDocumentUpdateDefinition<BsonDocument>(updateDocument);
    }
    else if (_update is BsonArray stages)
    {
        var pipeline = new BsonDocumentStagePipelineDefinition<BsonDocument, BsonDocument>(stages.Cast<BsonDocument>());
        updateDefinition = new PipelineUpdateDefinition<BsonDocument>(pipeline);
    }